File: app/src/main/java/com/kieran/winnipegbusbackend/StopFeature.java
package com.kieran.winnipegbusbackend;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

public class StopFeature implements Serializable {
    private static final String NAME_TAG = "name";
    private static final String COUNT_TAG = "count";
    private int count;
    private String name;

    public StopFeature(int count, String name) {
        this.count = count;
        this.name = name;
    }

    public StopFeature(JSONObject featureNode) {
        try {
            name = featureNode.getString(NAME_TAG);
            count = featureNode.getInt(COUNT_TAG);
        } catch (JSONException e) {
            //Intentionally blank because occasionally Winnipeg Transits API leaves out some fields
        }
    }

    public int getCount() {
        return count;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/StopFeature.kt
package com.kieran.winnipegbusbackend

import org.json.JSONException
import org.json.JSONObject

import java.io.Serializable

class StopFeature : Serializable {
    var count: Int = 0
        private set
    var name: String? = null
        private set

    constructor(count: Int, name: String) {
        this.count = count
        this.name = name
    }

    constructor(featureNode: JSONObject) {
        try {
            name = featureNode.getString(NAME_TAG)
            count = featureNode.getInt(COUNT_TAG)
        } catch (e: JSONException) {
            //Intentionally blank because occasionally Winnipeg Transits API leaves out some fields
        }

    }

    companion object {
        private val NAME_TAG = "name"
        private val COUNT_TAG = "count"
    }
}
